libp2p/muxers/mplex/mplex.nim

```diff
@@ -105,22 +107,16 @@ method handle*(m: Mplex) {.async, gcsafe.} =
                                      chann_iod = channel.oid
 
             if not isNil(m.streamHandler):
-              let stream = newConnection(channel)
-              m.conns.add(stream)
-              stream.peerInfo = m.connection.peerInfo
-              stream.observedAddr = m.connection.observedAddr
-
               var fut = newFuture[void]()
               proc handler() {.async.} =
                 try:
-                  await m.streamHandler(stream)
+                  await m.streamHandler(channel)
                   trace "finished handling stream"
-                  # doAssert(stream.closed, "connection not closed by handler!")
+                  # doAssert(channel.closed, "connection not closed by handler!")
                 except CatchableError as exc:
                   trace "exception in stream handler", exc = exc.msg
-                  doAssert(stream.closed, "stream not closed by protocol handler")
+                  doAssert(channel.closed, "stream not closed by protocol handler")
```

Comment:
this finally works fine now btw?

Reply:
actually no, we should probably reset the channel there either way